### PR TITLE
Fixed #17502. Zooming with blindness now changes blindness overlay size

### LINKED ISSUE #17502: zooming camera in/out doesnt increase/decrease the blindness radius
image

### PR DESCRIPTION
**Pull request by Plaba**
## About the PR

This PR fixes #17502.

**Media**

image

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl: plaba
- fix: Zooming with blindness now changes blindness circle size

### CODE REVIEW

File: Content.Client/Eye/Blinding/BlindOverlay.cs [hunk 1]
```diff
@@ -4,6 +4,7 @@
 using Robust.Shared.Enums;
 using Robust.Shared.Prototypes;
 using Content.Shared.Eye.Blinding;
+using Content.Shared.Movement.Components;
```
Comment (EmoGarbage404) [inline Content.Client/Eye/Blinding/BlindOverlay.cs:7]: idk if this is actually doing anything?

**Review by EmoGarbage404 — APPROVED** [review]
Looks fine at a glance. The screenshot in the pr proves it works so that's good enough for my (meaningless) approval.

File: Content.Client/Eye/Blinding/BlindOverlay.cs (hunk 1 above)
Comment (Plaba) [reply Content.Client/Eye/Blinding/BlindOverlay.cs]: fixed :)

File: Resources/Textures/Shaders/circle_mask.swsl [hunk 2]
```diff
@@ -1,17 +1,21 @@
+uniform highp vec2 ZOOM_SCALE;
+
 highp vec4 circle(in highp vec2 uv, in highp vec2 pos, highp float rad, in highp vec3 color) {
     highp float d = length(pos - uv) - rad;
     highp float t = clamp(d, 0.0, 1.0);
     return vec4(color, t);
 }
 void fragment(){
 	highp vec2 uv = FRAGCOORD.xy;
+    highp vec2 uv_stretch = vec2(uv.x * ZOOM_SCALE.x, uv.y * ZOOM_SCALE.y);
```
Comment (ElectroJr) [inline Resources/Textures/Shaders/circle_mask.swsl:10]: you should be able to just multiply vectors directly here and elsewhere
I.e., just use `= uv * ZOOM_SCALE;`

**Comment by ElectroJr** [comment]
This bug affects the crit overlay circles as well

**Comment by Plaba** [comment]
Would it be better to have the zoom effect (and any other spatial distortions) as shaders so that overlays do not have to care about zoom level?

**Comment by crazybrain23** [comment]
> Would it be better to have the zoom effect (and any other spatial distortions) as shaders so that overlays do not have to care about zoom level?

This would probably help with the singulo distortion being based on zoom level.

**Comment by metalgearsloth** [comment]
@Plaba just make electro's change and we can do the rest later (if you're still working on this).

**Pull request closed without merging**